spring-cloud-azure-feature-management/src/main/java/com/microsoft/azure/spring/cloud/feature/manager/feature/filters/TimeWindowFilter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.feature.manager.feature.filters;
+
+import static com.microsoft.azure.spring.cloud.feature.manager.FilterParameters.TIME_WINDOW_FILTER_SETTING_END;
+import static com.microsoft.azure.spring.cloud.feature.manager.FilterParameters.TIME_WINDOW_FILTER_SETTING_START;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import com.microsoft.azure.spring.cloud.feature.manager.FeatureFilter;
+import com.microsoft.azure.spring.cloud.feature.manager.entities.FeatureFilterEvaluationContext;
+
+/**
+ * A feature filter that can be used at activate a feature based on a time window.
+ */
+@Component("TimeWindowFilter")
+public class TimeWindowFilter implements FeatureFilter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TimeWindowFilter.class);
+
+    /**
+     * Evaluates whether a feature is enabled based on a configurable time window.
+     * @param context The feature evaluation context.
+     * @return True if the feature is enabled, false otherwise.
+     */
+    @Override
+    public boolean evaluate(FeatureFilterEvaluationContext context) {
+        String start = (String) context.getParameters().get(TIME_WINDOW_FILTER_SETTING_START);
+        String end = (String) context.getParameters().get(TIME_WINDOW_FILTER_SETTING_END);
+
+        ZonedDateTime now = ZonedDateTime.now();
+
+        if (!StringUtils.hasText(start) && !StringUtils.hasText(end)) {
+            LOGGER.warn("The {} feature filter is not valid for feature {}. It must specify either {}, {}, or both.",
+                    this.getClass().getSimpleName(), context.getName(), TIME_WINDOW_FILTER_SETTING_START,
+                    TIME_WINDOW_FILTER_SETTING_END);
+            return false;
+        }
+
+        ZonedDateTime startTime = ZonedDateTime.parse(start, DateTimeFormatter.RFC_1123_DATE_TIME);
```

Comment:
We assume that the start and end are always not null here? If only of them is not specified in context, what will be returned at line 37 or 38?

https://docs.oracle.com/javase/8/docs/api/java/time/ZonedDateTime.html#parse-java.lang.CharSequence-java.time.format.DateTimeFormatter-

Reply:
I thought I had tested this given the return, but it doesn't seem to work. Added a check.